cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlannerTest.java

```diff
@@ -119,19 +120,70 @@ public void testGetLeaderMovementTasks() {
                                           Collections.emptySet());
     StrategyOptions strategyOptions = new StrategyOptions.Builder(expectedCluster).build();
 
+    // 1: there should be no throttle if there is enough concurrency at both cluster level and broker level.
+    Map<Integer, Integer> leadershipMovementConcurrencyPerBroker = Map.of(
+        0, MAX_BROKER_CONCURRENCY,
+        1, MAX_BROKER_CONCURRENCY,
+        2, MAX_BROKER_CONCURRENCY,
+        3, MAX_BROKER_CONCURRENCY);
+    proposals = new ArrayList<>();
+    proposals.add(_leaderMovement1);
+    proposals.add(_leaderMovement2);
+    proposals.add(_leaderMovement3);
+    proposals.add(_leaderMovement4);
+    ExecutionTaskPlanner planner = new ExecutionTaskPlanner(null, new KafkaCruiseControlConfig(props));
     planner.addExecutionProposals(proposals, strategyOptions, null);
-    List<ExecutionTask> leaderMovementTasks = planner.getLeadershipMovementTasks(2);
-    assertEquals("2 of the leader movements should return in one batch", 2, leaderMovementTasks.size());
-    assertEquals(4, leaderMovementTasks.get(0).executionId());
+    List<ExecutionTask> leaderMovementTasks = planner.getLeadershipMovementTasks(leadershipMovementConcurrencyPerBroker, 4);
+    assertEquals("4 of the leader movements should return in one batch", 4, leaderMovementTasks.size());
+    assertEquals(leaderMovementTasks.get(0).proposal(), _leaderMovement1);
+    assertEquals(leaderMovementTasks.get(1).proposal(), _leaderMovement2);
+    assertEquals(leaderMovementTasks.get(2).proposal(), _leaderMovement3);
+    assertEquals(leaderMovementTasks.get(3).proposal(), _leaderMovement4);
+
+    // 2: task is throttled by cluster concurrency:
```

Comment:
Optional nit: Better practice is splitting these to different `@Test` method.  Since you're explaining the cases, I think they probably deserve a test case with good naming explaining

Reply:
Yeah I think that makes sense most of the cases. In cruise-control executor related tests, I see the tests classes tends to group many scenarios together in a single test. I feel the reason is there are quite a lot different branches needs to be tested for each method, which would potential cause the test number to easily 4X to 5X if separate each test branch. Another potential reason is setting up each test takes sometime, and we don't want the unit tests to spend much more time, as it is already very slow now.